core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -826,21 +824,34 @@ public SearchSourceBuilder scriptField(String name, Script script, boolean ignor
      */
     public SearchSourceBuilder indexBoost(String index, float indexBoost) {
         if (this.indexBoost == null) {
-            this.indexBoost = new ObjectFloatHashMap<>();
+            this.indexBoost = new ArrayList<>();
         }
-        this.indexBoost.put(index, indexBoost);
+        this.indexBoost.add(new Tuple<>(index, indexBoost));
         return this;
     }
 
     /**
      * Gets the boost a specific indices will receive when the query is
      * executed against them.
      */
-    public ObjectFloatHashMap<String> indexBoost() {
+    public List<Tuple<String, Float>> indexBoost() {
         return indexBoost;
     }
 
     /**
+     * Sets the boost indices will receive when the query is executed
+     * against it.
+     * This replaces existing boosts with new value.
+     *
+     * @param indexBoost
+     *            The boosts to apply to the indices
+     */
+    public SearchSourceBuilder indexBoost(List<Tuple<String, Float>> indexBoost) {
```

Comment:
see comment above, ideally this setter wouldn't be needed, as it's here only because we need it internally to replace the provided indices with their resolved names.